### PR TITLE
Camera freeze debug mode

### PR DESCRIPTION
**Pull request by FrancescoC-unity**
### Purpose of this PR
Create a debug mode for cameras to freeze them for the sake of culling. 

---
### Testing status
**Katana Tests**:  https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=camera-freeze-debug-mode&automation-tools_branch=add-platform-filter&unity_branch=trunk [Running]

---
### Overall Product Risks
**Technical Risk**: Low

**Halo Effect**: Low

---
### Comments to reviewers
I am not 100% convinced by the name I gave to the option in debug menu, I am very open to suggestions for better names!

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/Material/Lit/ShaderGraph/HDLitPass.template [hunk 1]
```diff
@@ -195,6 +195,7 @@ $include("SharedCode.template.hlsl")
         $SurfaceDescription.IridescenceMask:            surfaceData.iridescenceMask =           surfaceDescription.IridescenceMask;
         $SurfaceDescription.IridescenceThickness:       surfaceData.iridescenceThickness =      surfaceDescription.IridescenceThickness;
         $SurfaceDescription.RefractionIndex:            surfaceData.ior =                       surfaceDescription.RefractionIndex;
+        if ()
```
Comment (FrancescoC-unity) [inline com.unity.render-pipelines.high-definition/Editor/Material/Lit/ShaderGraph/HDLitPass.template]: Leftover from different change, will revert

**Comment by FrancescoC-unity** [comment]
Changed the UI and the way this works. The UI looks like this now:

image

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs [hunk 2]
```diff
@@ -964,13 +970,33 @@ protected override void Render(ScriptableRenderContext renderContext, Camera[] c
...
                     if (!camera.TryGetCullingParameters(camera.stereoEnabled, out cullingParams)) // Fixme remove stereo passdown?
                     {
                         renderContext.Submit();
                         continue;
                     }
 
+                    if(m_DebugDisplaySettings.IsCameraFreezeEnabled())
+                    {
+                        bool cameraIsFrozen = camera.name.Equals(m_DebugDisplaySettings.GetFrozenCameraName());
+                        if(cameraIsFrozen && !frozenCullingParamAvailable)
```
Comment (FrancescoC-unity) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs]: Yuck, let me write this in a better way

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs [hunk 3]
```diff
@@ -210,6 +210,10 @@ public int GetDecalAtlasMipCount()
 
         ComputeBuffer m_DepthPyramidMipLevelOffsetsBuffer = null;
 
+
+        ScriptableCullingParameters frozenCullingParams;
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs]: Please add a comment to explain what it is use for :)

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs [hunk 4]
```diff
@@ -964,13 +970,32 @@ protected override void Render(ScriptableRenderContext renderContext, Camera[] c
...
 
+
                     ScriptableCullingParameters cullingParams;
                     if (!camera.TryGetCullingParameters(camera.stereoEnabled, out cullingParams)) // Fixme remove stereo passdown?
                     {
                         renderContext.Submit();
                         continue;
                     }
 
+                    if(m_DebugDisplaySettings.IsCameraFreezeEnabled())
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs]: Please add a comment to explain what "freeze rendering is"

**Pull request merged**